WordPress/gutenberg
# Template field: Remove dependency with the "settings" object

## What?
Resolves #67521.

Improve the Template field in the post-edit screen by removing its dependency on block editor settings and fetching template data directly from core-data.

## Why?

- The current implementation of the Template field requires wrapping with ExperimentalBlockEditorProvider and depends on block editor settings, which creates unnecessary complexity and hidden dependencies.
- This change addresses the issue described in [Gutenberg Issue #67521](https://github.com/WordPress/gutenberg/issues/67521), where the Template field should work independently without requiring specific provider wrapping.

## How?

- Created a custom useTemplates hook to fetch template records directly from core-data
- Removed dependency on ExperimentalBlockEditorProvider
- Modified the Template field's Edit component to receive templates as a prop
- Simplified the template data fetching process
- Ensured the Template field can be used without additional context or settings

## Testing Instructions

1. Open the post/page editor
2. Navigate to the post settings sidebar
3. Verify the Template dropdown appears and is populated
4. Select different templates and ensure they can be applied
5. Test with both single and multiple post editing scenarios
